rfd/0180-sso-mfa.md

```diff
@@ -0,0 +1,852 @@
+must provide a client callback URL. Since this will only happen in out of date
+client, the error message will indicate that the user must upgrade to v17+.
+
+If the Proxy server is out of date, the MFA token will be lost at the end of
+the flow due to the SSO callback endpoints not knowing about the MFA token
+field. The client will catch when the MFA token field is missing and return an
+error indicating that the proxy server does not support SSO MFA, and to contact
+cluster administrators to update to v17+ if this is unexpected.
+
+### Audit Events
+
+SSO MFA checks will be tracked through the existing audit events that contain
+`MFADeviceMetadata`, using the [SSO MFA device](#mfa-device) tied to the SSO
+MFA session.
+
+In practice, the metadata will be filled out like this:
+- `DeviceName` - Name of the auth connector
+- `DeviceID` - UUID of the auth connector
+- `DeviceType` - SSO type (`OIDC`, `SAML`, or `Github`)
+
+```proto
+message MFADeviceMetadata {
+  // Name is the user-specified name of the MFA device.
+  string DeviceName = 1 [(gogoproto.jsontag) = "mfa_device_name"];
+  // ID is the UUID of the MFA device generated by Teleport.
+  string DeviceID = 2 [(gogoproto.jsontag) = "mfa_device_uuid"];
+  // Type is the type of this MFA device.
+  string DeviceType = 3 [(gogoproto.jsontag) = "mfa_device_type"];
+}
+```
+
+### Additional considerations
+
+#### OIDC ACR Values
+
+ACR values can be provided to an OIDC provider in an auth request to specify a
+specific type of authentication to perform. This can be useful for Teleport to
+specify to the IdP that MFA authentication is required.
+
+However, there are no common ACR values supported by all OIDC providers. Each
+provider will support its own arbitrary list of ACR values, if any at all.
+
+For example, Okta supports a phishing resistant (phr) acr value that would
+require Fido2/WebAuthn authentication to satisfy the requirement.
+
+Since this will vary between providers and configurations, Teleport will not
+use any ACR values by default, though we will document how to set `acr_values`
+in the OIDC connector in cases where it is useful.
+
+#### SAML RequestedAuthnContext
+
+A SAML client can provide `RequestedAuthnContext` to request a specific type of
+authentication. Similar to ACR values, the supported values vary between
+providers and configurations, so Teleport cannot make direct use of them.
+
+Once we find a SAML provider with a supported `RequestedAuthnContext` similar to
+Okta's phr ACR value, we may add a `requested_authn_context` field to the SAML
+connector resource to support it in certain configurations.
+
+Note: When using ADFS or JumpCloud as a SAML IdP, Teleport requires password
+auth by setting `PasswordProtectedTransport` as a minimum `RequestedAuthnContext`.
+This minimum will be skipped when the SAML connector is enabled for MFA only.
+
+#### SSO MFA device locking
+
+SSO MFA device locking will be left out of scope.
```

Comment:
Yes that could be possible as well, but it would not cover the scope of MFA device locking. IIUC in Teleport you can lock an MFA device to lock out ongoing sessions that have the MFA device's UUID set in the MFA verified certs.